tap_db2/__init__.py

```diff
@@ -56,83 +55,134 @@
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
+# Define data types
+
+# Full list
+#BIGINT - i
+#BLOB - ignore for now
+#CHARACTER - s
+#CLOB - ignore for now
+#DATE - d
+#DECIMAL - f
+#DOUBLE - f
+#INTEGER - i
+#SMALLINT - i
+#TIMESTAMP - d
+#VARCHAR - s
+#XML - s
+
 STRING_TYPES = set(
     [
-        "binary",
         "char",
-        "enum",
-        "longtext",
-        "mediumtext",
-        "nchar",
-        "nvarchar",
-        "text",
-        "uniqueidentifier",
-        "varbinary",
+        "character",
         "varchar",
+        "xml",
     ]
 )
 
 BYTES_FOR_INTEGER_TYPE = {
-    "tinyint": 1,
     "smallint": 2,
-    "mediumint": 3,
-    "int": 4,
     "integer": 4,
-    "real": 4,
+    "int": 4,
     "bigint": 8,
 }
 
-FLOAT_TYPES = set(
+# REAL -37
+# DOUBLE -307
+# DECFLOAT(16/8) -383
+# DECFLOAT(34/16) -6143
+# For practicality this will be limited to -50
+MAX_EXPONENT=-50
+
+FLOAT_TYPE_EXPONENT = {
+  "real":-37,
+  "double":MAX_EXPONENT,
+  "decfloat":MAX_EXPONENT,
+}
+
+DECIMAL_TYPES = set(
     [
-        "decfloat",
-        "double",
-        "float",
-        "money",
+        "decimal",
+        "numeric"
     ]
 )
 
 DATETIME_TYPES = set(
     [
-        "date",
-        "datetime",
-        "datetime2",
-        "smalldatetime",
-        "time",
         "timestamp",
     ]
 )
 
-VARIANT_TYPES = set(["json"])
+DATE_TYPES = set(
```

Comment:
Given we support the date datetype, we should support the time datatype as well like tap-mssql.

Reply:
Here are the lines for that.

https://github.com/wintersrd/pipelinewise-tap-mssql/blob/master/tap_mssql/__init__.py#L83
https://github.com/wintersrd/pipelinewise-tap-mssql/blob/master/tap_mssql/__init__.py#L139-L145
https://github.com/wintersrd/pipelinewise-tap-mssql/blob/master/tap_mssql/__init__.py#L139-L145

Reply:
Ok - that's been done